cfme/web_ui/mixins.py

```diff
@@ -27,7 +29,10 @@ def add_tag(tag, single_value=False, navigate=True):
             fill_d["category"] = "{} *".format(tag.category.display_name)
         else:
             fill_d["category"] = tag.category.display_name
-    fill(tag_form, fill_d)
+    try:
+        fill(tag_form, fill_d)
+    except NoSuchElementException:
```

Comment:
Have you audited all of the uses of this mixin to ensure that nothing is dependent on this raising a `NoSuchElementException`? 

Same applies below for changing the TypeError to an ItemNotFound error.

I like the idea of a consistent exception raised in these cases, but we can't change this mixin without accounting for all of its uses throughout the framework.